Source/DafnyLanguageServer/Util/PathExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.Dafny.LanguageServer.Language;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using System.IO;
+
+namespace Microsoft.Dafny.LanguageServer.Util
+{
+  /// <summary>
+  /// Collection of methods related to path operations to ensure that all path operations
+  /// are compatible with each other.
+  /// </summary>
+  public static class PathExtensions {
+    /// <summary>
+    /// Gets the path of the file represented by the given dafny document. The path returned is
+    /// in the standard system format. E.g. C:\data\test.dfy for windows or /home/test.dfy for linux.
+    /// </summary>
+    /// <param name="document">The document to get the file path of.</param>
+    /// <returns>The file path.</returns>
+    public static string GetFilePath(this DafnyDocument document) {
+      return GetFilePath(document.Uri);
+    }
+
+    /// <summary>
+    /// Gets the path of the file represented by the given text document. The path returned is
+    /// in the standard system format. E.g. C:\data\test.dfy for windows or /home/test.dfy for linux.
+    /// </summary>
+    /// <param name="document">The document to get the file path of.</param>
+    /// <returns>The file path.</returns>
+    public static string GetFilePath(this TextDocumentItem document) {
+      return GetFilePath(document.Uri);
+    }
+
+    private static string GetFilePath(DocumentUri uri) {
+      // GetFileSystemPath() is used since Path resolves to a non-Windows path format on Windows, e.g.:
+      // /d:/data/file.dfy
+      return uri.GetFileSystemPath();
+    }
+
+    /// <summary>
+    /// Checks if the given token is part of the entrypoint document.
+    /// </summary>
+    /// <param name="program">The dafny program to check the token against.</param>
+    /// <param name="token">The token to check.</param>
+    /// <returns><c>true</c> if the given token is part of the entrypoint document of the given program.</returns>
+    public static bool IsPartOfEntryDocument(this Dafny.Program program, Boogie.IToken token) {
+      // TODO Cleanup this check. It requires that DafnyLangParser sets the program's name to the entrypoint filename.
```

Comment:
I'm not sure leaving these TODO's here will add value. I suggest you resolve or remove them.

Reply:
I've removed the TODO and added a note to *DafnyLangParser*.